### PR TITLE
Add details on how the ClockMock::register works

### PR DESCRIPTION
**Pull request by moroine**
Hi,

I got some trouble mocking using the ClockMock feature. After dig into the code, I figure myself how it works ! 

So I decided to do a PR, but I'm not sure of the formatting,

### CODE REVIEW

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -231,6 +231,59 @@ test::
...
+        
+            $my = new MyClass();
+
+            $result = $my->getTimeInHours();
+
+            $this->assertEquals(time() / 3600, $result);
+        }
+    }
+    
+    class MyClass
```
Comment (OskarStark) [inline components/phpunit_bridge.rst]: no need to add this again, or remove the code sample above

Comment (moroine) [reply components/phpunit_bridge.rst]: True ! Thanks, I removed it

**Review by ndench — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by Jean85 — APPROVED** [review]

**Comment by moroine** [comment]
@ndench @OskarStark @Jean85 any update ?

**Comment by Jean85** [comment]
Build is broken, maybe we should re-trigger it to see if now is ok?

**Review by HeahDude — CHANGES_REQUESTED** [review]
Thanks for your work on this.

**Comment by moroine** [comment]
Thanks for reviewing, just updated according to your feedbacks

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -232,6 +232,51 @@ test::
...
 
+.. caution::
+
+    The ``@group time-sensitive`` annotation is equivalent to ``ClockMock::register(MyTest::class)``,
+    so if you want to get a time based function mocked into one of the source code you will need to
+    add it explicitly using ``ClockMock::register(MyClass::class)``. The ``ClockMock::register`` method
+    only create a mock of the time based functions into the same namespace as your class. So when using
+    ``time()`` you will use the mock instead of the default one.
+
+
```
Comment (HeahDude) [inline components/phpunit_bridge.rst]: Extra empty line can be removed.

File: components/phpunit_bridge.rst [hunk 3]
```diff
@@ -232,6 +232,51 @@ test::
...
+.. code-block:: php
```
Comment (HeahDude) [inline components/phpunit_bridge.rst]: Can be replaced by `::` at the end of line 241 instead of the dot.

File: components/phpunit_bridge.rst [hunk 4]
```diff
@@ -232,6 +232,51 @@ test::
...
+
+    class MyClass
+    {
+        public function getTimeInHours()
+        {
+            return time() / 3600;
+        }
+    }
+
+.. code-block:: php
```
Comment (HeahDude) [inline components/phpunit_bridge.rst]: All the code added here should keep an extra indent to be part of the caution block.

**Review by HeahDude — COMMENTED** [review]
Looks good, thanks! last comments from me.

File: components/phpunit_bridge.rst (hunk 4 above)
Comment (moroine) [reply components/phpunit_bridge.rst]: So I can indent also the code-block ?

File: components/phpunit_bridge.rst (hunk 3 above)
Comment (HeahDude) [reply components/phpunit_bridge.rst]: Indeed, that's what I meant. Fon't bother you can push and let our host hook build the demo, its link will be accessible in the PR checks.

File: components/phpunit_bridge.rst (hunk 4 above)
Comment (HeahDude) [reply components/phpunit_bridge.rst]: Yes!

File: components/phpunit_bridge.rst (hunk 3 above)
Comment (moroine) [reply components/phpunit_bridge.rst]: Unfortunately, it isn't deployed on Platform.sh as specified here https://symfony.com/doc/current/contributing/documentation/overview.html#review-your-changes while branch 2.8 is in a maintained branch.

File: components/phpunit_bridge.rst [hunk 5]
```diff
@@ -232,6 +232,48 @@ test::
 
 And that's all!
 
+.. caution::
+
+    The ``@group time-sensitive`` annotation is equivalent to ``ClockMock::register(MyTest::class)``,
+    so if you want to get a time based function mocked into one of the source code you will need to
```
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: time-based (missing dash)
I don't understand "into one of the source code". Maybe "into another class"?

File: components/phpunit_bridge.rst [hunk 6]
```diff
@@ -232,6 +232,48 @@ test::
...
+    add it explicitly using ``ClockMock::register(MyClass::class)``. The ``ClockMock::register`` method
+    only create a mock of the time based functions into the same namespace as your class. So when using
```
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: creates (missing "s") - not need for "only" I suppose
time-based

**Comment by nicolas-grekas** [comment]
@moroine it could be great to add a note to fix #9620 also btw, could you have a look? This seems very related to me.

**Comment by moroine** [comment]
Done

**Review by HeahDude — APPROVED** [review]
Thanks!

**Comment by moroine** [comment]
@nicolas-grekas  @Jean85 I think it's ready to be merged :angel:

**Comment by javiereguiluz** [comment]
@moroine thanks for improving this doc! I'm sorry it took us so long to merge it. Also, congrats on your first Symfony Docs contribution!

**Pull request closed without merging**